Remove enterprise plans, but add enterprise toggle

corywatilo opened the pull request:
- Starts updating pricing page based on [this mock](https://www.figma.com/file/CLj2U34xpNiHuZRo73GJPm/posthog.com?node-id=5282%3A42027)
 - I started creating the plan grid at the top, and was moving bits from the previous version (down below) to this new version at top.
 - The plans look good, and the responsiveness is 🤌🏼 with some fancy grid reordering

## To-dos

- [x] Move pricing calculator to right column
- [x] Add toggle for _Enterprise mode_ in right rail, adjust pricing in calculator (and in plans estimates) to match
- [x] We still have 4 CTAs. CTAs should change when _Enterprise mode_ is activated
    - [x] Add text to plan titles to reflect the change (PostHog Cloud -> PostHog Enterprise Cloud, Self-hosted -> Self-hosted Enterprise)
- [x] Schedule a demo - need to figure out CTA, but we're basically removing the logic for group demos
- [x] Get in touch - likely a new contact form?

corywatilo commented:
Closes [product-internal/#331](https://github.com/PostHog/product-internal/issues/331)

corywatilo commented:
## Final todos

- [ ] With _Enterprise mode_ ON, "Free" should say "Included
    <img width="347" alt="image" src="https://user-images.githubusercontent.com/154479/183105402-85f2f412-8b24-42c7-b37b-f301fdb6cc05.png">
- [ ] Add `$` in front of all event pricing (except "Free", "Included" obviously)
- [ ] Enterprise pricing in the calculator (at the 2 million mark) is incorrect. (Showing at $450 when it $450 is the price for 1 mil. 450 + (.00045*1000000) = **$900**
    <img width="1131" alt="image" src="https://user-images.githubusercontent.com/154479/183105622-0be9029a-c687-4662-a3fb-e4245161cc7d.png">

timgl commented:
Last week one of our conclusions was that people would literally spend 20 minutes playing around with the various options on the old pricing page, leading them to not sign up. This looks cool, but this will reintroduce the same issue i suspect!

corywatilo commented:
I'm actually not convinced _that_ was the reason, as much as the fact we didn't have a grid of plans _at all_. It just told you which plan you should choose (after answering a couple questions) without the ability to let the visitor browse and decide for themself. So it was hard to tell how many permutations there were at a glance.

Why I see this as being different is we're actually showing "all" (2) plans, then you can toggle the single option we have on both.

That said, it'd be easy to bump the pricing up, hide the calculator under a hidden div, and have it expand below the pricing.

### Collapsed 

<img width="1217" alt="image" src="https://user-images.githubusercontent.com/154479/183149616-e0cc2a13-a862-43e2-b642-d72cffc04e6e.png">

### Expanded

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/154479/183149752-3bbe8b02-4e2d-4ac8-804e-061e3692d91c.png">

But IMO, I'd say it's worth trying this, and A/B testing with hiding it.

But as a whole, I think the feel of this page is entirely different than the wizard-only approach which didn't work out so well.

timgl commented:
Yep I think this makes sense. Definitely let’s A/B test.

Couple of other small things:
- That orange border looks like it means you can click on the self hosted panel, but (at least on iPad) you can’t.
- The two panel plans have a white background, even in dark mode. In general there are a couple of issues in dark mode for the pricing page.
- I think we should put the “PostHog open source” section just under the current plans, based on feedback like this: https://posthoghelp.zendesk.com/agent/tickets/329 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/3402)

(Sorry for late response)

corywatilo commented:
> In general there are a couple of issues in dark mode for the pricing page.

Hmm we don't actually have a dark mode for the pricing page - curious how you ended up there? (I tried toggling on from a content page that has the toggle and visited again but no luck reproducing.)

## Changes

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/154479/184174293-ab93a526-3c6f-4532-b20b-b002fcdf3581.png">

I updated this to:

1. Add a "badge" on the recommended plan (to make it appear less clickable)
2. Moved pricing breakdown to top of right column, calculator section to bottom
3. Moved Open Source plan callout to just below the paid plans
4. Added Open Source as a column to the plan comparison

## To-dos

Halp @smallbrownbike 

- [ ] In the comparison section, because I added Open Source, we now have an uneven number of columns in the CSS grid, and because it's componentized, I can't control having 2 columns for Cloud and 3 for self-host. Could use help getting this section to have 5 even columns.

    <img width="1077" alt="image" src="https://user-images.githubusercontent.com/154479/184174832-cdba752b-70ee-4974-ad5d-f1b4916dd123.png">

- [ ] I borked the pricing in that section for non-enterprise plans. It should show the currently-live pricing like this (plus Open Source should show **Free**)

    <img width="1208" alt="image" src="https://user-images.githubusercontent.com/154479/184175091-7d6991ac-ca63-4c84-9bdf-4a1d4484ce9b.png">

- [ ] Set up experiment for showing/hiding the pricing calculator by default

    1. One version will show the calculator...

        <img width="394" alt="image" src="https://user-images.githubusercontent.com/154479/184175408-f6bce470-0f98-4e75-bc9e-c7a3520fd6ac.png">

    2. ... and the other won't. (In `Calculator/index.tsx:278`, you'll see it's all set up to be displayed onclick, and we need the experiment set up around this too.)

        <img width="389" alt="image" src="https://user-images.githubusercontent.com/154479/184175902-c17edb35-a959-4c99-942f-39d0744fdb5d.png">

        <img width="390" alt="image" src="https://user-images.githubusercontent.com/154479/184175960-7943cc0d-5e81-4e46-9e4f-ae002fe230cf.png">

jamesefhawkins commented:
Minor feedback:

- if we're a/b testing, I'd suggest we have a version with pricing table _and_ pricing slider all collapsed by default

corywatilo commented:
> pricing table and pricing slider all collapsed by default

If this is all in the sidebar but collapsed, then we're showing no pricing, other than free for 1 mil events. In this case (when collapsed) would you suggest we show the first even price tier in the plans instead, until the full pricing table is shown? (Otherwise feels more plans-focused and not pricing.)

jamesefhawkins commented:
@corywatilo could we have a quick play around today some point when you're free for a call? I'm not tottally sure what you mean
